app/src/main/java/com/gusta/wakemehome/utilities/NotificationUtils.java

```diff
@@ -8,24 +9,117 @@
 import android.graphics.BitmapFactory;
 import android.graphics.Color;
 import android.os.Build;
+import android.view.View;
 
 import androidx.core.app.NotificationCompat;
 import androidx.core.app.TaskStackBuilder;
 
+import com.google.android.material.snackbar.Snackbar;
 import com.gusta.wakemehome.MainActivity;
 import com.gusta.wakemehome.R;
 
 public final class NotificationUtils {
 
+    //================//
+    // PUBLIC METHODS //
+    //================//
+
     /**
-     * Constructs and displays a notification.
+     * Show a text message to the user.
+     *
+     * @param context The context asking to show the message.
+     * @param text    The message text.
+     */
+    public static void notifyUser(Context context, String text) {
+        notifyUser(context, text, null, null, null, null);
+    }
+
+    /**
+     * Show a text message to the user.
+     *
+     * @param activity           The activity to show the message on.
+     * @param mainTextStringId   The id for the string resource for the message's main text.
+     * @param actionTextStringId The id for the string resource for the text of the action item.
+     * @param listener           The listener associated with the Snackbar action.
+     */
+    public static void notifyUser(Activity activity, int mainTextStringId, int actionTextStringId,
+                                  View.OnClickListener listener) {
+        notifyUser(activity, activity.getString(mainTextStringId), null,
```

Comment:
which activity will be send on service?

Reply:
A service can't ask for permissions as it can't show the dialog box (actually it can but it's bad user experience)